### PR TITLE
Introduced ACL.lambda convenience method

### PR DESCRIPTION
**Pull request by jglick**
Self-contained, but filing for discussion to see whether reviewers consider this type of addition desirable—if so, it could be extended to other similar cases in core and plugins.

Java 8 lambdas can automatically be cast to single-abstract-method (SAM) `interface`s, but for reasons discussed in the expert group, not to `abstract class`es. This makes it awkward for `-source 8` code to implement otherwise simple interfaces such as `ACL`, which are logically just a single function with a type signature and documentation—i.e., what could have been a `@FunctionalInterface` with `default` methods had that been an option when the API was written. By introducing an adapter method taking a SAM (such as from `java.util.function.*`), we can make it possible to write code which is almost as concise as an implicit cast to a SAM.

Proposed changelog entry:
* Added convenience API method `ACL.lambda`.

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by daniel-beck** [comment]
Looks reasonable, although I wonder about the naming.

File: core/src/main/java/hudson/security/ACL.java [hunk 1]
```diff
@@ -95,6 +96,15 @@ public final boolean hasPermission(@Nonnull Permission p) {
      */
     public abstract boolean hasPermission(@Nonnull Authentication a, @Nonnull Permission permission);
 
+    public static ACL lambda(final BiFunction<Authentication, Permission, Boolean> impl) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/security/ACL.java]: Some javadoc would be useful

**Review by oleg-nenashev — DISMISSED** [review]
Documentation should be extended, but LGTM in general

**Review by oleg-nenashev — APPROVED** [review]
🐝

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Pull request merged**